api/grpc/server/server.go

```diff
@@ -56,6 +57,9 @@ func (s *apiServer) CreateContainer(ctx context.Context, c *types.CreateContaine
 	e.Runtime = c.Runtime
 	e.RuntimeArgs = c.RuntimeArgs
 	e.StartResponse = make(chan supervisor.StartResponse, 1)
+	if runt.GOOS == "solaris" && len(c.Id) >= 12 {
```

Comment:
We currently have a limitation on the length of the container ID on Solaris. The permissible length is currently lesser than sha256 size. We're temporarily using the first 12 of the ID (12 is arbitrary, the max permissible is much larger than that). This is consistent all the way from the Docker engine store to the runtime.